source/extensions/filters/http/transformation/transformation_filter_config.cc

```diff
@@ -24,6 +26,202 @@ TransformerSharedPtr Transformation::getTransformer(
+    stages_[i] = std::move(temp_stages[i]);
+  }
+}
+
+void PerStageRouteTransformationFilterConfig::addTransformation(const envoy::api::v2::filter::http::RouteTransformations_Transformations& transformation) {
+  using envoy::api::v2::filter::http::RouteTransformations_Transformations;
+  // create either request or response one.
+  switch (transformation.transformation_case()) {
+    case RouteTransformations_Transformations::kRequestMatch: {
+      auto&& request_match = transformation.request_match();
+      TransformerConstSharedPtr request_transformation;
+      TransformerConstSharedPtr response_transformation;
+      Matcher::MatcherConstPtr matcher;
+
+      if (request_match.has_match()){
+        matcher = Matcher::Matcher::create(request_match.match());
+      }
+
+      bool clear_route_cache = request_match.clear_route_cache();
+      if (request_match.has_request_transformation()) {
+        try {
+          request_transformation = Transformation::getTransformer(
+              request_match.request_transformation());
+        } catch (const std::exception &e) {
+          throw EnvoyException(
+              fmt::format("Failed to parse request template: {}", e.what()));
+        }
+      }
+      if (request_match.has_response_transformation()) {
+        try {
+          response_transformation = Transformation::getTransformer(
+              request_match.response_transformation());
+        } catch (const std::exception &e) {
+          throw EnvoyException(
+              fmt::format("Failed to parse response template: {}", e.what()));
+        }
+      }
+
+      if (request_transformation != nullptr || response_transformation != nullptr) {
+
+        TransformerPairConstSharedPtr transformer_pair =
+            std::make_unique<TransformerPair>(request_transformation,
+                                              response_transformation,
+                                              clear_route_cache);
+        transformer_pairs_.emplace_back(matcher, transformer_pair);
+      }
+      break;
+    }
+    case RouteTransformations_Transformations::kResponseMatch:{
+      auto&& response_match = transformation.response_match();
+      ResponseMatcherConstPtr matcher;
+      if (response_match.has_match()) {
+        matcher = ResponseMatcher::create(response_match.match());
+      }
+      auto&& transformation = response_match.response_transformation();
+        try {
+          std::pair<ResponseMatcherConstPtr, TransformerConstSharedPtr> pair(std::move(matcher), Transformation::getTransformer(transformation));
+          response_transformations_.emplace_back(std::move(pair));
+        } catch (const std::exception &e) {
+          throw EnvoyException(
+              fmt::format("Failed to parse response template on response matcher: {}", e.what()));
+        }
+      break;
+    }
+    case RouteTransformations_Transformations::TRANSFORMATION_NOT_SET: {
+      // TODO: log warning?
```

Comment:
lets log a warning here

Reply:
will add validation so this fails configuration checks and will add assert